pkg/packagekit/updates.jsx

```diff
@@ -449,57 +555,259 @@ class ApplyUpdates extends React.Component {
+            </Flex>
+        );
+
+        const title = (<>
+            <RebootingIcon />
+            <span className="update-success-table-title">
+                {cockpit.format(cockpit.ngettext("$0 package needs a system reboot", "$0 packages need a system reboot", reboot.length), reboot.length)}
+            </span>
+        </>);
+
+        entries.push({
+            columns: [
+                { title },
+            ],
+            props: { key: "reboot" },
+            hasPadding: true,
+            expandedContent,
+        });
+    }
+
+    if (restart.length > 0) {
+        const col1 = restart.slice(0, Math.round(restart.length / 2));
+        const col2 = restart.slice(Math.round(restart.length / 2));
+        const expandedContent = (
+            <Flex>
+                <FlexItem flex={{ default: 'flex_1' }}>
+                    <TextContent>
+                        <TextList>
+                            {col1.map(service => (<TextListItem key={service}>{service}</TextListItem>))}
+                        </TextList>
+                    </TextContent>
+                </FlexItem>
+                {col2.length > 0 && <FlexItem flex={{ default: 'flex_1' }}>
+                    <TextContent>
+                        <TextList>
+                            {col2.map(service => (<TextListItem key={service}>{service}</TextListItem>))}
+                        </TextList>
+                    </TextContent>
+                </FlexItem>}
+            </Flex>
+        );
+
+        const title = (<>
+            <ProcessAutomationIcon />
+            <span className="update-success-table-title">
+                {cockpit.format(cockpit.ngettext("$0 service needs to be restarted", "$0 services need to be restarted", restart.length), restart.length)}
+            </span>
+        </>);
+
+        entries.push({
+            columns: [
+                { title },
+            ],
+            props: { key: "service" },
+            hasPadding: true,
+            expandedContent,
+        });
+    }
+
+    if (manual.length > 0) {
+        const col1 = manual.slice(0, Math.round(manual.length / 2));
+        const col2 = manual.slice(Math.round(manual.length / 2));
+        const expandedContent = (
+            <Flex>
+                <FlexItem flex={{ default: 'flex_1' }}>
+                    <TextContent>
+                        <TextList>
+                            {col1.map(service => (<TextListItem key={service}>{service}</TextListItem>))}
+                        </TextList>
+                    </TextContent>
+                </FlexItem>
+                {col2.length > 0 && <FlexItem flex={{ default: 'flex_1' }}>
+                    <TextContent>
+                        <TextList>
+                            {col2.map(service => (<TextListItem key={service}>{service}</TextListItem>))}
+                        </TextList>
+                    </TextContent>
+                </FlexItem>}
+            </Flex>
+        );
```

Comment:
This code appears three times. Please split it out into a `TwoColumnFlexList` (or similar name) component.